### PR TITLE
Refactor to allow publish support

### LINKED ISSUE #10: [FR] Add Publish.css
This is for me, to be able to reference when I add the PR.

### PR DESCRIPTION
**Pull request by sigrunixia**
Lets fix this thing up!

This will close #10

### CODE REVIEW

**Comment by dbarenholz** [comment]
Hi Sig!

Can you please split this up into the following PRs? 

1. Addition of publish support. Also think about modifying the build script so it automatically builds when running `npm run build`.
2. Addition of stylelint, so that I can modify whatever stylelint wants, if desired. 
3. _After_ (2) is merged: the refactor (potentially - I still have a hard time looking at the 3000+ lines changed). 

Thank you!

**Comment by dbarenholz** [comment]
Also: any reason to bump the obsidian version? Is the theme not compatible with 1.0.0 anymore?

**Comment by sigrunixia** [comment]
> Also: any reason to bump the obsidian version? Is the theme not compatible with 1.0.0 anymore?

We're going to be updating the sample templates to start at 1.5., so doing this for parity.

**Comment by dbarenholz** [comment]
> > Also: any reason to bump the obsidian version? Is the theme not compatible with 1.0.0 anymore?
> 
> We're going to be updating the sample templates to start at 1.5., so doing this for parity.

https://docs.obsidian.md/Plugins/User+interface/Right-to-left it appears we'll have to put it at 1.6 and make it RTL compatible :)

**Comment by sigrunixia** [comment]
If I recall correctly, we actually do not have any of the "faults" in Halcyon.

**Comment by sigrunixia** [comment]
I also suggest we do a gentle refactor first. If Publish goes first, I'd have to redo the work again.

Here is what ebullient and I did for splitting the files up into Publish and not-publish. Would this work for you?

https://github.com/ebullient/obsidian-theme-ebullientworks/tree/main/src/fragments

**Comment by dbarenholz** [comment]
> I also suggest we do a gentle refactor first. If Publish goes first, I'd have to redo the work again.
> 
> Here is what ebullient and I did for splitting the files up into Publish and not-publish. Would this work for you?
> 
> https://github.com/ebullient/obsidian-theme-ebullientworks/tree/main/src/fragments

The idea: yes.
The implementation of the idea (so a single `fragments` folder, and all files being dumped in there): no.

I want to keep separate files according to the documentation as listed in <https://docs.obsidian.md/Reference/CSS+variables/CSS+variables>.

I'm happy to refactor first; does the fragment-idea work while keeping separate files according to the link above?

File: scss/editor/_headings.scss [hunk 1]
```diff
@@ -37,10 +43,4 @@ body:is(.theme-dark) {
 	// --h4-variant: ;
 	// --h5-variant: ;
 	// --h6-variant: ;
-	--h1-weight: 700;
-	--h2-weight: 675;
-	--h3-weight: 650;
-	--h4-weight: 625;
-	--h5-weight: 600;
-	--h6-weight: 575;
```
Comment (dbarenholz) [inline scss/editor/_headings.scss]: I would like to keep these in order as they appear in the documentation: https://docs.obsidian.md/Reference/CSS+variables/Editor/Headings

I see that there's new variables, but that's for a different PR.

File: scss/editor/_horizontal-rule.scss [hunk 2]
```diff
@@ -1,9 +1,4 @@
 // Horizontal rule
-body:is(.theme-dark) {
-	--hr-color: color-mix(
-			in hsl shorter hue,
-			var(--text-faint),
-			var(--background-modifier-border)
-		)
-		// --hr-thickness: ;;
```
Comment (dbarenholz) [inline scss/editor/_horizontal-rule.scss]: I'd like to keep these in the same order as the documentation. Please don't remove them, even if they're unused.

File: scss/editor/_link.scss [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 	// --link-decoration: ;
 	// --link-decoration-hover: ;
 	// --link-decoration-thickness: ;
-	--link-unresolved-color: var(--halcyon-palette-lilac);
 	// --link-unresolved-opacity: ;
 	// --link-unresolved-filter: ;
-	--link-unresolved-decoration-style: dashed;
-	--link-unresolved-decoration-color: var(--halcyon-palette-lilac);
-	--link-external-color: var(--halcyon-palette-lilac);
-	--link-external-color-hover: var(--halcyon-palette-lilac-hover);
```
Comment (dbarenholz) [inline scss/editor/_link.scss:14]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_properties.scss [hunk 4]
```diff
@@ -23,11 +26,8 @@ body:is(.theme-dark) {
 	// --metadata-label-font-size: ;
 	// --metadata-label-font-weight: ;
 	// --metadata-sidebar-label-font-size: ;
-	--metadata-label-text-color: var(--halcyon-palette-cyan);
-	--metadata-label-text-color-hover: var(--halcyon-palette-cyan-hover);
```
Comment (dbarenholz) [inline scss/editor/_properties.scss]: I'd like to keep these in the same order as the documentation.

File: scss/components/_tabs.scss [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
+  --tab-text-color-active: var(--text-accent);
+  // --tab-text-color-focused:
+  --tab-text-color-focused-active: var(--text-accent);
+  // --tab-text-color-focused-highlighted:
+  --tab-text-color-focused-active-current: var(--text-accent);
+  // --tab-font-size
+  // --tab-font-weight
+  // --tab-container-background:
+  --tab-divider-color: var(--titlebar-background);
+  --tab-outline-color: var(--halcyon-accent);
```
Comment (dbarenholz) [inline scss/components/_tabs.scss:14]: Is there a reason why this is no longer commented?

File: scss/components/_toggle.scss [hunk 6]
```diff
@@ -1,8 +1,8 @@
 // Toggle
-body:is(.theme-dark) {
+.theme-dark body {
 	--toggle-border-width: 10px;
-	// --toggle-width: 1px;
-	// --toggle-radius: 0px;
+	--toggle-width: 1px;
+	--toggle-radius: 0px;
```
Comment (dbarenholz) [inline scss/components/_toggle.scss:5]: Is there a reason why this is no longer commented?

File: scss/core-plugins/_file-explorer.scss [hunk 7]
```diff
@@ -1,6 +1,6 @@
 // File explorer
-body:is(.theme-dark) {
+.theme-dark body {
 	--vault-name-font-size: 1em;
-	// --vault-name-font-weight: ;
```
Comment (dbarenholz) [inline scss/core-plugins/_file-explorer.scss]: I'd like to keep these in the same order as the documentation.

File: scss/core-plugins/_graph.scss [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
+	--graph-line: var(--halcyon-base-blue-03);
+	--graph-node-unresolved: var(--halcyon-palette-salmon);
+	--graph-node-tag: var(--halcyon-palette-lilac);
 	// --graph-controls-width: ;
 	// --graph-text: red;
-	--graph-line: var(--halcyon-base-blue-03);
 	// --graph-node: ;
-	--graph-node-unresolved: var(--halcyon-palette-salmon);
 	// --graph-node-focused: ;
-	--graph-node-tag: var(--halcyon-palette-lilac);
```
Comment (dbarenholz) [inline scss/core-plugins/_graph.scss]: I'd like to keep these in the same order as the documentation.

File: scss/core-plugins/_search.scss [hunk 9]
```diff
@@ -1,8 +1,8 @@
 // Search
-body:is(.theme-dark) {
+.theme-dark body {
 	--search-clear-button-color: var(--halcyon-palette-salmon);
-	// --search-clear-button-size: ;
 	--search-icon-color: var(--halcyon-accent);
-	// --search-icon-size: 10px;
+	--search-icon-size: 10px;
```
Comment (dbarenholz) [inline scss/core-plugins/_search.scss:5]: Is there a reason why this is no longer commented?

File: scss/core-plugins/_search.scss [hunk 10]
```diff
@@ -1,8 +1,8 @@
 // Search
-body:is(.theme-dark) {
+.theme-dark body {
 	--search-clear-button-color: var(--halcyon-palette-salmon);
-	// --search-clear-button-size: ;
 	--search-icon-color: var(--halcyon-accent);
-	// --search-icon-size: 10px;
```
Comment (dbarenholz) [inline scss/core-plugins/_search.scss]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_blockquote.scss [hunk 11]
```diff
@@ -1,8 +1,5 @@
 // Blockquote
-body:is(.theme-dark) {
-	--blockquote-color: color-mix(
-		in srgb,
-		var(--halcyon-palette-yellow) 75%,
-		var(--halcyon-text-normal)
-	);
+.theme-dark body {
+	--blockquote-color: color-mix(in srgb, var(--halcyon-palette-yellow) 75%, var(--halcyon-text-normal));
+	--callout-padding: var(--size-4-3) var(--size-4-6);
```
Comment (dbarenholz) [inline scss/editor/_blockquote.scss:4]: I believe this `--callout-padding` should not be here.

File: scss/editor/_callout.scss [hunk 12]
```diff
@@ -1,8 +1,8 @@
 // Callout
-body:is(.theme-dark) {
+.theme-dark body {
+	--callout-padding: var(--size-4-3) var(--size-4-6);
 	// --callout-border-width: ;
 	// --callout-border-opacity: ;
-	--callout-padding: var(--size-4-3) var(--size-4-6);
```
Comment (dbarenholz) [inline scss/editor/_callout.scss]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_callout.scss [hunk 13]
```diff
@@ -25,19 +25,20 @@ body:is(.theme-dark) {
...
+			}
+
+			margin: 1em 0;
+			padding: var(--callout-padding);
+		}
+		// Fix: codeblocks in callouts in live preview
+		> pre {
+			margin: 1em 0;
+		}
+	}
```
Comment (dbarenholz) [inline scss/editor/_callout.scss:43]: I understand the rewrite for `.theme-dark body {}`, but why the rest?

Also: there is an empty `.callout` block, is this intended?

File: scss/editor/_code.scss [hunk 14]
```diff
@@ -1,8 +1,6 @@
 // Code
-body:is(.theme-dark) {
+.theme-dark body {
 	--code-background: var(--halcyon-base-blue-01);
-	// --code-white-space
-	// --code-size
```
Comment (dbarenholz) [inline scss/editor/_code.scss]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_lists.scss [hunk 15]
```diff
@@ -1,13 +1,14 @@
 // Lists
-body:is(.theme-dark) {
-	// --list-indent: ;
-	// --list-spacing: ;
```
Comment (dbarenholz) [inline scss/editor/_lists.scss]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_table.scss [hunk 16]
```diff
@@ -1,15 +1,15 @@
...
+	--table-header-size: larger;
 	// --table-border-width: ;
 	// --table-border-color: var(--halcyon-palette-cyan);
 	// --table-white-space: ;
 	// --table-header-background:  var(--background-secondary);
-	--table-header-background-hover: var(--table-background);
 	// --table-header-border-width: ;
 	// --table-header-border-color: ;
 	// --table-header-font: ;
-	--table-header-size: larger;
```
Comment (dbarenholz) [inline scss/editor/_table.scss]: I'd like to keep these in the same order as the documentation.

File: scss/editor/_tags.scss [hunk 17]
```diff
@@ -1,20 +1,12 @@
...
-	--tag-background: color-mix(
-		in hsl shorter hue,
-		var(--tag-color) 25%,
-		transparent
-	);
-	--tag-background-hover: color-mix(
-		in hsl shorter hue,
-		var(--tag-color-hover) 25%,
-		transparent
-	);
```
Comment (dbarenholz) [inline scss/editor/_tags.scss]: I'd like to keep these in the same order as the documentation.

File: scss/extra/_fix-border-radius.scss
Comment (dbarenholz) [inline scss/extra/_fix-border-radius.scss:1]: Very nice, thanks for the scss magic!

File: scss/plugins/_dataview.scss [hunk 18]
```diff
@@ -1,15 +1,20 @@
...
+//
 
-/*
-  - This plugin enables many "generated" views for your note metadata.
-  - One of those are tables, which by default have no border.
-*/
+//
+//  - This plugin enables many "generated" views for your note metadata.
+//  - One of those are tables, which by default have no border.
+//
```
Comment (dbarenholz) [inline scss/plugins/_dataview.scss:10]: nitpick: For multiline comments I personally prefer the `/* */` style.

File: scss/plugins/_highlightr-plugin.scss [hunk 19]
```diff
@@ -1,95 +1,70 @@
...
-*/
+  //
+  //github: https://github.com/chetachiezikeuzor/Highlightr-Plugin
+  //id: highlightr-plugin
+  //support added on: 29/09/2023
+  //
+  //- This plugin allows for various highlights using inline <mark> elements.
+  //- These have default colors, which I have set to use Halcyon colors.
+  //- There are also some "padding/margin" fixes
+  //  for the various highlighting styles that the plugin provides.
```
Comment (dbarenholz) [inline scss/plugins/_highlightr-plugin.scss:9]: nitpick: For multiline comments I personally prefer the `/* */` style.

File: scss/plugins/_obsidian-dynamic-highlights.scss [hunk 20]
```diff
@@ -1,39 +1,44 @@
...
+//  id: obsidian-dynamic-highlights
+//  support added on: 29/09/2023
+//
+//
+//
+//  - This plugin dynamically highlights text based on selection.
+//  - It has a built-in editor for styles, which uses code mirror.
+//  - sailKite styled it to approximate Halcyon coding style
+//
+// Dynamic Highlights (by sailKite)
```
Comment (dbarenholz) [inline scss/plugins/_obsidian-dynamic-highlights.scss:12]: nitpick: For multiline comments I prefer the `/* */` style.

File: scss/plugins/_obsidian-dynamic-highlights.scss [hunk 21]
```diff
@@ -1,39 +1,44 @@
...
-/*
-  - This plugin dynamically highlights text based on selection.
-  - It has a built-in editor for styles, which uses code mirror.
-  - sailKite styled it to approximate Halcyon coding style 
-*/
+.theme-dark body {
+  .modal {
+    .cm-gutters {
+      color: var(--halcyon-accent);
+    }
```
Comment (dbarenholz) [inline scss/plugins/_obsidian-dynamic-highlights.scss:18]: Very nice, much more readable.

File: scss/window/_ribbon.scss [hunk 22]
```diff
@@ -1,7 +1,7 @@
 // Ribbon
-body:is(.theme-dark) {
-	// --ribbon-background: ;
```
Comment (dbarenholz) [inline scss/window/_ribbon.scss]: I'd like to keep these in the same order as the documentation.

File: scss/window/_window-frame.scss [hunk 23]
```diff
@@ -1,11 +1,11 @@
 // Window frame
-body:is(.theme-dark) {
+.theme-dark body {
 	--titlebar-background: var(--background-secondary);
 	--titlebar-background-focused: var(--background-secondary);
-	// --titlebar-border-width:
```
Comment (dbarenholz) [inline scss/window/_window-frame.scss]: I'd like to keep these in the same order as the documentation.

**Review by dbarenholz — CHANGES_REQUESTED** [review]
Gitub can now show me an actually diff (as opposed to "file is deleted") so I can properly review.

Some notes:
1. I'd like to keep variables in the same order as the documentation. Even if that means that there are used variables, some unused variables, and then more used variables. I find it easier to reference back to the docs if they're in the same order.
2. I like that you simplify selectors, and use scss magic where applicable.
3. I've noticed a few instances where you modified things -- can you comment on why this is done?

**Comment by dbarenholz** [comment]
@sigrunixia Many comments in the review are effectively "please move a line up or down" -- there's only like 3 instances that require thought.

**Comment by sigrunixia** [comment]
Closing to resubmit, because editing existing PR when your branch is gone is annoying.

**Pull request closed without merging**